subsys/bluetooth/host/Kconfig

```diff
@@ -96,10 +96,10 @@ config BT_RX_STACK_SIZE
 	depends on BT_HCI_HOST || BT_RECV_IS_RX_THREAD
 	default 1024
 	default 2048 if BT_MESH
-	default 512 if CONFIG_BT_HCI_RAW
-	range 1024 65536
+	default 512 if BT_HCI_RAW
+	range 512 65536 if BT_HCI_RAW
 	range 1100 65536 if BT_MESH
-	range 512 65536 if CONFIG_BT_HCI_RAW
+	range 1024 65536
 	help
```

Comment:
If I move back the default range to be the first one in the list CONFIG_BT_RX_STACK_SIZE gets set to 1024 (auto.conf). It only works as desired when I move it to be the last range option.

I believe this issue also happens with other options, but looking at subsys/bluetooth/host/Kconfig this probably got unnoticed because the default range is still enough to handle the default value for all the conditions.

Before:

    $ make BOARD=96b_carbon_nrf51 -C samples/bluetooth/hci_spi
    $ grep BT_RX_STACK_SIZE 
    samples/bluetooth/hci_spi/outdir/96b_carbon_nrf51/include/config/auto.conf
    CONFIG_BT_RX_STACK_SIZE=1024

With the patch:

    $ grep BT_RX_STACK_SIZE 
    samples/bluetooth/hci_spi/outdir/96b_carbon_nrf51/include/config/auto.conf
    CONFIG_BT_RX_STACK_SIZE=512